ToDoAppWithCoreData/Controllers/ListViewController.swift

```diff
@@ -0,0 +1,178 @@
+
+import UIKit
+import CoreData
+
+protocol ListViewControllerDelegate {
+    func tapSaveButton(item: String, detail: String, beginDate: Date, beginTime: Date, endDate: Date, endTime: Date, exitingItem: NSManagedObject?)
+}
+
+
+class ListViewController: UIViewController, UITextViewDelegate {
+    
+    public var item: String = ""
+    public var detail: String = ""
+    public var beginDate: Date!
```

Comment:
beginDateが必須なのであれば、DB上もnon-optionalにするべきかと
必須ではないならnilを入れた方が良いです。

```suggestion
    public var beginDate: Date? = nil
```